contracts/ZkVerifyAggregationIsmp.sol

```diff
@@ -0,0 +1,128 @@
+// SPDX-License-Identifier: Apache-2.0
+
+pragma solidity 0.8.20;
+
+import "@polytope-labs/ismp-solidity/interfaces/IIsmpModule.sol";
+import "@polytope-labs/ismp-solidity/interfaces/IIsmpHost.sol";
+import "@polytope-labs/ismp-solidity/interfaces/Message.sol";
+import "@polytope-labs/ismp-solidity/interfaces/IDispatcher.sol";
+import "./interfaces/IZkVerifyAggregation.sol";
+import "@openzeppelin/contracts/access/AccessControl.sol";
+import "./lib/Merkle.sol";
+
+/**
+ * @title ZkVerifyAggregationIsmp Contract
+ * @notice It allows receiving (from Hyperbridge), persisting and verifying aggregation proofs coming from zkVerify chain.
+ */
+contract ZkVerifyAggregationIsmp is IZkVerifyAggregation, AccessControl, BaseIsmpModule {
+
+    // IIsmpHost Address
+    address private _host;
+
+    /// @dev Role required for operator to submit/verify proofs.
+    bytes32 public constant OPERATOR = keccak256("OPERATOR");
+
+    /// @notice Latest valid aggregationId for bridge events.
+    uint256 public latestAggregationId;
+
+    /// @notice Mapping of MC aggregationIds to proofsAggregations.
+    mapping(uint256 => bytes32) public proofsAggregations;
+
+    bool public isEnforcingSequentialAggregations;
+
+    /// @notice Emitted when a new aggregation is posted.
+    /// @param _aggregationId Event aggregationId.
+    /// @param _proofsAggregation Aggregated proofs.
+    event AggregationPosted(uint256 indexed _aggregationId, bytes32 indexed _proofsAggregation);
+
+    /// @notice Posted _aggregation must be sequential.
+    error InvalidAggregation();
+
+    /// @notice Prevent owner from handing over ownership
+    error OwnerCannotRenounce();
+
+    /**
+     * @notice Construct a new ZkVerifyAggregationIsmp contract
+     * @param _operator Operator for the contract
+     * @param _operator Ismp host contract address
+    */
+    constructor(
+        address _operator,
+        address _ismpHost
+    ) {
+        _grantRole(DEFAULT_ADMIN_ROLE, msg.sender); // it is used as owner
+        _grantRole(OPERATOR, _operator);
+        _host = _ismpHost;
+    }
```

Comment:
I have one question. According to the [official Solidity docs](https://docs.soliditylang.org/en/v0.8.20/contracts.html#multiple-inheritance-and-linearization), the constructor of the base class is always called before the constructor of the derived class. In our case, this means that the constructor of `BaseIsmpModule` is called before the constructor of `ZkVerifyAggregationIsmp`.
The [constructor of `BaseIsmpModule`](https://github.com/polytope-labs/ismp-solidity/blob/2a358f8cfc2a0eecfab10f0b7fa2e02daca189dd/interfaces/IIsmpModule.sol#L111)
```solidity
constructor() {
	address hostAddr = host();
	if (hostAddr != address(0)) {
		// approve the host infintely
		IERC20(IDispatcher(hostAddr).feeToken()).approve(hostAddr, type(uint256).max);
	}
}
```
internally calls the `host()` function, defined as
```solidity
function host() public view override returns (address) {
    return _host;
}
```
Since the variable `_host` is initialized only later in the body of `ZkVerifyAggregationIsmp`, when the `host()` function gets called by `BaseIsmpModule` constructor, `_host` defaults to `address(0)`, and also `hostAddr` gets set equal to `address(0)`. In particular, this means that the code inside the if branch is not executed.
Is this intended? Is it relevant to our case?

Reply:
I committed a possible approach to solve the issue https://github.com/HorizenLabs/zkv-attestation-contracts/commit/6969fb4252af70304a07f0c5dfb5b0443b657bde

Reply:
You are right about this issue thanks for pointing it out! 
we could have approved the contract with another tx after deployment but no reason to not issue the already existing logic.

I added the code + add tests